client/src/components/AppHeader.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { useLocation, Link } from 'react-router-dom';
+import { Icon, Label, Button, Menu } from 'semantic-ui-react';
+
+import { LegacyAPI } from '../api';
+
+import * as Auth from './Auth';
+import { SemanticICONS } from 'semantic-ui-react/dist/commonjs/generic';
+
+export const AppHeader: React.FC<{}> = () => {
+    const { pathname } = useLocation();
+    const loom = '*';
+
+    // TODO: Hacky implementation. To be refactored/reviewed properly
+    const countAnnotations = (metadata) => {
+        let count = 0;
+        for (const clustering of metadata['cellMetaData']['clusterings']) {
+            for (const cluster of clustering['clusters']) {
+                count += cluster['cell_type_annotation'].length;
+            }
+        }
+        return count;
+    };
```

Comment:
You're right that this isn't great code (see the TODO) and that the `count` it computes is not usefully used. I am inclined to delete it outright because `pages/Annotations.tsx` should know how to handle `0` annotations without another component trying to hold its hand. @KrisDavie ?

Reply:
Having the menu item for annotation always be visible might also not be a bad thing for user experience.
It can become confusing if a menu item sometimes is visible and sometimes is not, without a clear indication of why this is.
Having the page itself handle missing annotations sounds indeed like a better approach.